src/Messaging/DomainMessage.php

```diff
@@ -106,52 +91,38 @@ protected function init()
         }
 
         if ($this->createdAt === null) {
-            $time = microtime(true);
+            $time = (string) microtime(true);
             if (false === strpos($time, '.')) {
                 $time .= '.0000';
             }
             $this->createdAt = \DateTimeImmutable::createFromFormat('U.u', $time);
         }
     }
 
-    /**
-     * @return Uuid
-     */
-    public function uuid()
+    public function uuid() : Uuid
     {
         return $this->uuid;
     }
 
     /**
      * @return int
      */
-    public function version()
+    public function version() : int
     {
         return $this->version;
     }
 
-    /**
-     * @return \DateTimeImmutable
-     */
-    public function createdAt()
+    public function createdAt() : \DateTimeImmutable
```

Comment:
@prolic @oqq we should fix `MessageDataAssertion` instead and not change the type hint of createdAt, see: https://github.com/prooph/common/blob/master/src/Messaging/Message.php#L46

`Message` should stay immutable.